File: authentication/src/main/java/authentication/exception/GlobalExceptionHandler.java
package authentication.exception;

import application.exception.CommonClientErrorCode;
import application.exception.UnAuthorizedException;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(UnAuthorizedException.class)
    public ResponseEntity<ApiErrorResponse> handleUnAuthorizedException(UnAuthorizedException e) {
        String errorMessage = e.getMessage();

        CommonClientErrorCode errorCode = CommonClientErrorCode.UNAUTHORIZED;
        ApiErrorResponse response = ApiErrorResponse.of(errorCode.getCode(), errorMessage, errorCode.getHttpStatus());

        return ResponseEntity.status(errorCode.getHttpStatus()).body(response);
    }
}

File: api/src/main/kotlin/api/global/exception/GlobalExceptionHandler.kt
package api.global.exception

import application.exception.CommonClientErrorCode
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.MethodArgumentNotValidException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(MethodArgumentNotValidException::class)
    fun handleMethodArgumentNotValidException(e: MethodArgumentNotValidException): ResponseEntity<ApiErrorResponse> {
        val errorMessage = e.bindingResult.fieldErrors
            .mapNotNull { it.defaultMessage }
            .joinToString(", ")
        val errorCode = CommonClientErrorCode.VALID_EXCEPTION
        val response = ApiErrorResponse.of(errorCode.code, errorMessage, errorCode.httpStatus)
        return ResponseEntity.status(errorCode.httpStatus).body(response)
    }

}